aws-datastore/src/main/java/com/amplifyframework/datastore/AWSDataStorePlugin.java

```diff
@@ -255,6 +261,30 @@ public void configure(
             event -> InitializationStatus.SUCCEEDED.toString().equals(event.getName()),
             event -> categoryInitializationsPending.countDown()
         );
+
+        reachabilityMonitor.configure(context);
+        observeNetworkStatus();
+    }
+
+    private void observeNetworkStatus() {
+        Amplify.Hub.subscribe(HubChannel.DATASTORE,
+            hubEvent -> hubEvent.getData() instanceof NetworkStatusEvent,
+            hubEvent -> {
+                if (((NetworkStatusEvent) hubEvent.getData()).getActive()) {
+                    LOG.info("Network gained");
+                    start(
+                        (Action) () -> { },
+                        ((e) -> LOG.error("Error starting datastore plugin after network event: " + e))
+                    );
+                } else {
+                    LOG.info("Network lost");
+                    start(
+                            (Action) () -> { },
+                            ((e) -> LOG.error("Error stopping datastore plugin after network event: " + e))
+                    );
+                }
```

Comment:
`start` is being called in both "Network gained" and "Network lost", is this correct?